### PR TITLE
Internationalization without hack errors

### PR DESCRIPTION
**Pull request by maydewd**

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by gsingh93** [comment]
Thanks for doing this @maydewd! @markszabo, this PR fixes some of the hack errors from your PR, so I'll be closing yours. It includes all of your commits though, so you'll still be getting all the credit for implementing this.

File: src/language/language.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+require_once($_SERVER['DOCUMENT_ROOT'] . '/../vendor/autoload.php');
+
+/* HH_IGNORE_ERROR[1002] */
+$lang = null;
+
+async function tr_start(): Awaitable<void> {
+  $config = await Configuration::gen('language');
+  $language = $config->getValue();
+  $document_root = must_have_string(Utils::getSERVER(), 'DOCUMENT_ROOT');
+  if(preg_match('/^\w{0,5}$/',$language) and file_exists($document_root . "/language/lang_".$language.".php")) {
```
Comment (gsingh93) [inline src/language/language.php]: nit: we're probably inconsistent about this in other places in the repo, but we should stick to Facebook convention and not put spaces around `.`.

Comment (gsingh93) [inline src/language/language.php]: Can you make the `\w{0,5}` into `\w{2}`? We should stick to ISO 639-1 convention for all our language files, where each language code is exactly two characters.

**Pull request merged**